cmp/models.py

```diff
@@ -103,18 +102,47 @@ def __str__(self):
 
 
 class SoldierDeath(models.Model):
-    soldier = models.OneToOneField(Soldier, on_delete=models.CASCADE, primary_key=True)
-    date = models.DateField(null=True, blank=True)
-    company = models.ForeignKey(Company, blank=True, null=True, default="UNKNOWN", on_delete=models.CASCADE) 
-    cemetery = models.ForeignKey(Cemetery, blank=True, null=True, default=110, on_delete=models.CASCADE)
+    soldier  = models.OneToOneField(Soldier, on_delete=models.CASCADE, related_name='soldierdeath')
+    date     = models.DateField(null=True, blank=True)
+    company  = models.ForeignKey(Company, blank=True, null=True, default="UNKNOWN", on_delete=models.CASCADE, related_name='companies')
+    cemetery = models.ForeignKey(Cemetery, blank=True, null=True, default=110, on_delete=models.CASCADE, related_name='cemeteries')
     cwgc_id  = models.IntegerField(blank=True, null=True, unique=False, verbose_name="War Graves ID")
-
-    def __str__(self):
-        return f"{self.soldier} {self.date} {self.cemetery}"
+    def __unicode__(self):
+        return '%s %s %s' % (self.Soldier, self.Date, self.cemetery)
```

Comment:
You may want to look at these model methods. Capital 'S' Soldier is not valid.

Reply:
while you're at it might be worth it to update these to use f-strings: `f"{self.soldier} {self.date} {self.cemetery}"`